Nakshatras and Rashis as stars, asterisms and constellations.

vvasuki opened the pull request:
We show Nakṣatra-s associated with a single star as a star, rather than as a constellation. Whether we show a given Nakṣatra as an asterism or as a constellation depends on whether we can associate the corresponding asterism with related and intuitive constellation art.

Also fixing a couple of typos.

vvasuki commented:
Tested, looking good!
![image](https://user-images.githubusercontent.com/2664797/72202594-710b0b80-3487-11ea-82f3-d905617b6837.png)

gzotti commented:
The skyculture system unfortunately so far has no differentation for the various Lunar stations found in Asian cultures. Ultimately, Nakshatras, Lunar Mansions etc. will be classified into yet another group. For the time being, their line art should be preferably coded as "asterisms", and other constellations as "constellations". At some later day developers may try to re-sort the items, based on the description and "invisible" documentation (README or inline info) given, into the respective classes. Probably, there should also be artwork entries for asterisms. If you have artwork now, is it useful to code those both as constellations and asterisms? (Not sure how that looks, will Nakshatra artwork collide with other constellation artwork?) Important would also be translation of translatable names into English, so that the meaning of some native name could be understood.

vvasuki commented:
Namaste!

To clarify:
- I understand the limitations of stellarium viz https://github.com/Stellarium/stellarium/issues/371 and https://github.com/Stellarium/stellarium/issues/437 .
- With this PR, we are not trying to implement divisions of the ecliptic/ lunar mansions (be it nakShatra-s or rAshi-s) in any way. We just want to be able to seamlessly identify the asterisms originally associated with them.
- All rAshi-s are/ were associated with constellations (esp. the 12 zodiacal ones you're familiar with). These were in the past marked as asterisms without skyart. That has been fixed, and skyart added/ restored..
- Some naxatra-s were associated with a single star (and no imagery can be defined/ intuited/ suggested by just a single star). These were marked as fake constellations. That is being fixed.
- Some naxatra-s were associated with an asterism with no clear associated imagery, but were marked as constellations. This is being fixed.
- Some naxatra-s were associated with an asterism as well as mythic imagery. These did not have associated art - that's being fixed.

Hope this clarifies - please let me if you have any other questions/ suggestions.

vvasuki commented:
Also, with regards to possible collisions between rAshi and naxatra artwork - this mostly does not happen, and can in a later date be resolved by prioritising one over the other (ie removing lines in constellationsart.fab) - but as of now, this is not a problem, and I rather sorely miss the constellation art which could be easily copied from the western skyculture.

alex-w commented:
Is your PR ready to review and merge?

vvasuki commented:
@alex-w Yes. Ready.

alex-w commented on skycultures/indian/constellation_names.eng.fab:
```diff
@@ -19,49 +20,53 @@
+# Provenance for the below: Apte dictionary, purana-s.
+UMa	"ṛkṣa"	_("Ursa Major")
+# Provenance - BG Tilak's book, veda-s (eg. Aitareya AraNyaka quoted above). Amarakosha says:  मृगशीर्षं मृगशिरस्तस्मिन्नेवाग्रहायणी। इल्वलास् तच्छिरोदेशे तारका निवसन्ति याः॥. Some modern dictionaries thence identify iNvakA-s with the head of Orion, but given its identification with the arrow, it is more likely to be Orion's belt as illustrated by Tilak. 
+iNvakA	"Invakā"	_("Arrow")
```
Need a put an EOL char (add new empty line)

gzotti replied:
Not only that. Please add at least approximate English translations to the Indian strings. You can explicitly state [approximately: ... translation: vvasuki]. 
For the provenance, a file references.fab (see chapter 9.11 in the User Guide) may be helpful.

The Indian skyculture is highly complex, and Stellarium currently is not able to map this complexity. I have the impression this is the third iteration of additions I am aware of, every time bringing new spelling variants to translate, however often without new translatable (understandable) content. 

What I dislike currently is a separation of the Nakshatras into asterisms or constellations "based on available sky art". This does not make any sense to me. If nakshatras deserve sky art in principle, they should all be defined as constellations, and sky art can be added also later. You can comment them as nakshatras so that when the time comes (see below), they will be separated out. The distinction into asterisms vs. constellations is a modern concept of "adding line art bits in addition to the canonical 88 constellations" that has been developed and works best in context of the Western skyculture.  
If needed, you can repeat all line art for nakshatras again into the asterisms file to allow "display asterisms==display nakshatras". Then of course don't add other asterisms to that file. 

We are in these months discussing extending the whole skyculture thing, but have not defined any new formats yet. I would like to make Lunar Stations and Zodiacal constellations two distinct groups which both should be displayable also with artwork. And of course, single-star "constellations" should be marked in some way when lines are active (just a circled star?).  But there are also still astronomical problems to solve. We just don't have enough developers.

vvasuki replied:
> Need a put an EOL char (add new empty line)

Done.

> Not only that. Please add at least approximate English translations to the Indian strings. You can explicitly state [approximately: ... translation: vvasuki].

I checked and found approximate english translations for constellation and star names. Was there something else you needed translated?

> For the provenance, a file references.fab (see chapter 9.11 in the User Guide) may be helpful.

Good idea - added. I feel it is important to **allow comments in each .fab file** - that way one can add line-specific comments. Currently, one cannot provide citation for a particular line-art or sky-art.

>  I have the impression this is the third iteration of additions I am aware of, every time bringing new spelling variants to translate, however often without new translatable (understandable) content.

Part of the problem was that a contributor (Tanmoy) comes from a region of India (Bengal) where they don't distinguish between v and b; and he did not show sufficient Sanskrit language and transliteration knowledge to accurately type the names. That's the only source of variation - otherwise, there is no dispute about the canonical transliterated spelling. 
I also don't fully understand the comment - could you please provide an example for " without new translatable (understandable) content" here?

> What I dislike currently is a separation of the Nakshatras into asterisms or constellations "based on available sky art". This does not make any sense to me. If nakshatras deserve sky art in principle, they should all be defined as constellations, and sky art can be added also later.

To clarify, *some* nakShatra-s deserve sky art (png files, rather than lines). Not all. It is not so much a question of availability as much as *existence*. Given this clarification, how do you recommend we proceed?

gzotti replied:
We are in a process of collecting and gathering requirements for better handling of skycultures, although we fundamentally lack development time (and, I openly admit, ethnological and cultural knowledge). This includes a coming separation of Lunar stations and probably Zodiacal constellations. This will not happen next week, but - time allowing - within the next few years. 

"Asterisms" were added in the last years to show what "modern" people understand as "inofficial" star group, like the Big Dipper, Summer Triangle or Winter Hexagon. Those figures currently don't show constellation art, although clearly a Big Dipper or Sickle could be shown with an artistic rendering, so this may also change, if somebody is willing to add them. Given the uncontrolled growth of asterisms, we should also find some way of selecting favourites, e.g. there are "Winter Hexagon" and "Big G" formed by practically the same stars, seeing any of the two is a personal preference.  

If inside a collection of entities there are entries with some common optional peculiarity, all entities should go into the class that allows such peculiarity. In our case, if *some* Nakshatras may receive sky art, *all* Nakshatras should be coded as constellations. When we do the change mentioned above, those will then be separated (or just marked inside the constellation file). I don't know whether the Indian traditional sky has non-constellational asterisms in a similar add-on concept as described above. If you want to have a way to "show all nakshatras as line art, while leaving the other constellations switched off", add all nakshatras additionally as the only possible asterisms. Then you can, for now, switch on "asterisms" to show the "nakshatras". 

Currently we have no way to show a 1-star constellation/asterism. A workaround would be a very short line to the closest (also dim!) HIP star, so that at least a label gets placed next to the star, and write in the comments that the dim star is not official part of that constellation. We may highlight the 1-star constellation in later editions e.g. by a circle. 

We should indeed be able to add comment handling (just ignore everything past a `#` sign) very soon. After the comment sign, we may collect extra data per line, but have not decided about the formats yet. You see the listing of references in the star_names.fab. We may use a similar thing for the other lists. So, the first string after the `#` can be a comma-separated list of references.

On "without translatable (understandable) content": e.g.
> pUrvAShADhA	_("Pūrva aṣādhā", "asterism")
> uttarAShADhA	_("Uttara aṣādhā", "asterism")

Does Asadha mean anything? Or Purva and Uttara? 

You are using full (transcribed) names as abbreviation for the constellations. The "abbreviated" mode should allow to use short unique markers, without filling the screen with text. In scientific context, the 88 IAU constellations are abbreviated with 3 characters, which is just enough to remember the constellation. I am sure you can find such shorter names. The old codes N01, N02...  were also OK because they marked the nakshatras. 

You deleted the month names from the description. Were they wrong?

vvasuki commented:
- Restored naxatra-s to constellations list.
- Fixed abbreviations, added translations.

> You deleted the month names from the description. Were they wrong?

Not wrong, just not very relevant. Nakshatra-s are associated with all sorts of things ranging from rituals and person names to month names (arising from ancient association with full moon days). One can read up on it in one of the references - not a good idea to clutter description.

vvasuki commented:
Ping..

suchakr commented:
The additional line and sky arts proposed by this PR are useful enhancements.  I am based in Bangalore and use both the desktop and mobile Stellarium for inquiry and discussions with fellow archeo-astronomy enthusiasts - for tracing celestial object positions in past dates based on information in ancient Indic ( Hindu/Jaina/Bauddha, etc. ) literature. These changes will add richness to our discussions without having to go through the tech pains of an ad-hoc installation of sky-culture files.   The inclusion of these changes in the desktop version and if possible the mobile version would delight our community of Stellarium and archeo-astronomy enthusiasts.

alex-w commented:
@suchakr is sky culture complete now?

@gzotti Georg, could you review it?

anandg commented on skycultures/indian/constellation_names.eng.fab:
```diff
@@ -19,49 +20,85 @@
+# From @gzotti: "if some Nakshatras may receive sky art, all Nakshatras should be coded as constellations. When we do the change mentioned above, those will then be separated (or just marked inside the constellation file). ... We may highlight the 1-star constellation in later editions e.g. by a circle." 
 
-## We've referred to highly regarded sanskrit to sanskrit dictionaries kalpadruma and vAchaspatyam for etymologies below.
+Aśv	"Aśvinī"	_("Divine physician twins")
```
minor comment, personal preference: I find it odd to see diacritical marks mixed with first letter capital (though i can still read it unambiguously). my preference is for small-caps with diacritics.

vvasuki replied:
I myself am ambivalent - but standard practice seems to be to capitalize initial letter of proper nouns.
Thanks for drawing my attention there though - Ensured that initial letter is always capital for consistency.

anandg commented on skycultures/indian/constellation_names.eng.fab:
```diff
@@ -19,49 +20,85 @@
-N09	"Aśleṣā"		_("The embrace")
-N10	"Maghā"			_("The bountiful one")
-# TRANSLATORS: phālgun - Red
-N11	"Pūrba phālgunī"	_("The former red one")
```
Thanks for updating this and other entries with a standardised spelling (Pūrba -> Pūrva).

anandg reviewed (commented):
Thanks @vvasuki for a nice set of changes. All of us in the Indic astronomy community are happy to see such efforts. @gzotti - please consider approving this PR. It looks like a simple set of changes, and it'll encourage more of us in the astro-software community to make ongoing contributions.

gzotti commented on skycultures/indian/constellation_names.eng.fab:
```diff
@@ -19,49 +20,85 @@
-## We've referred to highly regarded sanskrit to sanskrit dictionaries kalpadruma and vAchaspatyam for etymologies below.
+Aśv	"Aśvinī"	_("Divine physician twins")
+Bhar	"Bharanī"	_("Bearer")
+Kṛt	"Kṛttikā"	_("Cutter")
```
I see severely different translations for some names given (Ardra: moist or wet? Krttika??? phalguni=red (just the color), or fruitful, as in "ripe (-->red?) fruit"; chitra=just bright or now even spectacular?; svati=honest, or untranslatable proper name? Dhanista=wealthiest, or most famous???), and even a swap in sequence,  as e.g. former N05 Mrgasira has moved down. Was there a reason for changing the sequence? I cannot say what is the best translation, but the fruitful/red example shows what simple translation of a word without context is missing! It may help to give some extended information in tabular form in the description.en.utf8, e.g. if there are different regional traditions?

vvasuki replied:
Namaste @gzotti - 

## Regarding translations

The prior translations (as well as wrong spellings) came without any provenance - in some cases reflecting contributor shrI tanmoy's unfamiliarity with the sanskrit language. Sanskrit being highly standardized since ancient, does not recognize big regional varieties. In this changelist, we refer to authoritative dictionaries when needed. For example, in case of phAlgunI, a dictionary derivation (from kalpadruma) is given in the comment. In case of dhaniSThA, the translation "most famous" is likely derived from an alternative name for the same, shraviShThA - so that the english rendering did not match the traditional name. As to why phAlgunI would be "fruitful" or "maghA" bountiful would require much speculation based on vedic rituals and lore - a succinct context is not handily available.

Change in sequence is an accident of moving items to and fro between asterisms and nakshatra-s in the course of this CL. I will restore the original order.

vvasuki replied:
(Fixed order.)

gzotti commented:
Thank you for the updates and explanations. 

I can not really review the quality of the new version, but I should also not stand in the way of merging if there is a consensus that this is better than it used to be. 

However, please update info.ini and description.en.utf8 to reflect your coauthorship, and maybe add contact info when Stellarium has added the mentioned technical requirements that would invite further changes.

t4saha commented:
Naksatra and Rashi are overlapping with each other hence I used one of them as asterism in the original construction. Naksatra is more likely moon mansions hence it realy deserves a separate sky culture. 

For english translation i haven't use the literal meaning in many cases as the alternative names linked to many mythology and season are very exciting. And the names are also personified and feminised as they are considered as spouses of the moon.

Ex - 
1. Kṛttikā (pleiades), literally means cutter but it also means six nurses of son of lord Siva.  (And pleiades has 6 visible star.)

2. Rohini is the most loved wife of the moon, hence she is the "spirited one".

3. Punarvasu, he is the restorer of good, truth etc. (taken from mythology associated with prajapati , the creator, and the mother of God's)

4. Phalguni means color red and fruitful both as it is also related to spring season (some texts marks it as ending of a year and point of a new beginning)

5. Ardra means both moist and wet (i don't know about the current standardization). I used moist because it resembles softness, wet is like crying.

Etc...

And finally we should create new arts for these nakstras and rashi.

Thank for updating the sky culture.

vvasuki commented:
> 
> However, please update info.ini and description.en.utf8 to reflect your coauthorship, and maybe add contact info when Stellarium has added the mentioned technical requirements that would invite further changes.

Done (by linking to github page)!

> Thank for updating the sky culture with new names and constellation along with detail description.

Thanks for clarifying and contributing, Tanmoy. Thanks also to anandg and suchakr - with future active contribution, I'm sure we'll continue to enrich experience of Hindu (archeo-)astronomy enthusiasts.

gzotti commented:
@alex-w I think we can merge this now for 0.20.3, and have possible further development and PRs in the future.
Thanks to all contributors!

vvasuki commented:
Looked more closely at the etymologies and updated.
